pkg/util/memory/tracker.go

```diff
@@ -800,15 +801,24 @@ func (t *Tracker) Reset() {
 	t.mu.children = nil
 }
 
-func (t *Tracker) getParent() *Tracker {
-	t.parMu.Lock()
-	defer t.parMu.Unlock()
-	return t.parMu.parent
+func (t *Tracker) getParent() (tracker *Tracker) {
```

Comment:
How about changing
```
	parMu struct {
		parent *Tracker // The parent memory tracker.
		sync.Mutex
	}
```
to `parent atomic.Pointer[Tracker]`

Reply:
> `parMu` can be removed(mutex is useless) and only keep `parent atomic.Pointer[Tracker]`.

`parMu` is removed now.